### PR TITLE
Update storybook and switch it to webpack5

### PR DESCRIPTION
**Pull request by iAdramelk**
New storybook enables webpack5 and esm support, so I updated to them to make future esm migration and unfied-* dependencies update easier.

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -6,14 +6,16 @@ const COMPANY_LOGOS_DIRECTORY = resolve(__dirname, "..", "components/Company");
 
 module.exports = {
   stories: ["../components/**/*.stories.mdx", "../components/**/*.stories.tsx"],
-  addons: [
-    "@storybook/addon-links",
-    "@storybook/addon-essentials",
-    "@storybook/addon-actions",
-  ],
+  addons: ["@storybook/addon-essentials", "storybook-addon-turbo-build"],
```
Comment (iAdramelk) [inline .storybook/main.js:9]: `addon-essentials` already includes `addon-actions` so we can remove it and we do not use `addon-links` anywhere.

`storybook-addon-turbo-build` make build somewhat faster. We may make it a lot faster after we remove `babel` and switch to `swc`. But to do what we need `swc` plugin for `styled-components`. `nextjs` team is working on it now.

File: .storybook/main.js [hunk 2]
```diff
@@ -6,14 +6,16 @@ const COMPANY_LOGOS_DIRECTORY = resolve(__dirname, "..", "components/Company");
...
+  core: {
+    builder: "webpack5",
+  },
   webpackFinal: async (config) => {
-    config.resolve.plugins.push(new TsconfigPathsPlugin({}));
+    // This plugin will make `baseUrl` prop from tsconfig.json
+    // work with DocGen plugin, without it wit will throw erros
```
Comment (C-STYR) [inline .storybook/main.js:15]:
```suggestion
    // work with DocGen plugin, without it it will throw errors
```

File: .storybook/preview.js [hunk 3]
```diff
@@ -3,6 +3,11 @@ import theme from "components/theme";
 import "components/global-styles.css";
 import * as NextImage from "next/image";
 
+// NextImage will not work with styorybook out of the box
```
Comment (C-STYR) [inline .storybook/preview.js:6]:
```suggestion
// NextImage will not work with Storybook out of the box
```

Comment (C-STYR) [inline .storybook/preview.js]:
```suggestion
// NextImage will not work with Storybook out of the box
```

Comment (C-STYR) [inline .storybook/preview.js]:
```suggestion
// NextImage will not work with Storybook out of the box
```

**Review by C-STYR — APPROVED** [review]

**Review by Alqanar — APPROVED** [review]

**Pull request merged**